core/jobs/batch_jobs/collection_info_jobs.py

```diff
@@ -0,0 +1,123 @@
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Jobs that extract Collection models information."""
+
+from __future__ import annotations
+
+from core.jobs import base_jobs
+from core.jobs.io import ndb_io
+from core.jobs.transforms import job_result_transforms
+from core.jobs.types import job_run_result
+from core.platform import models
+
+import apache_beam as beam
+
+from typing import Iterable, Tuple
+
+MYPY = False
+if MYPY: # pragma: no cover
+    from mypy_imports import collection_models
+    from mypy_imports import feedback_models
+    from mypy_imports import user_models
+
+(
+    collection_models, feedback_models, user_models
+) = models.Registry.import_models([
+    models.NAMES.collection, models.NAMES.feedback, models.NAMES.user])
+
+datastore_services = models.Registry.import_datastore_services()
+
+
+class GetCollectionOwnersEmailsJob(base_jobs.JobBase):
+    """Job that extracts collection id and user email from datastore."""
+
+    def run(self) -> beam.PCollection[job_run_result.JobRunResult]:
+
+        collection_pair = (
+            self.pipeline
+            | 'get collection models ' >> ndb_io.GetModels(
+                collection_models.CollectionRightsModel.get_all(
+                    include_deleted=False))
+            | 'Flatten owner_ids and format' >> beam.ParDo(
+                FlattenIDs())
+        )
+
+        user_pair = (
+            self.pipeline
+            | 'Get all user settings models' >> ndb_io.GetModels(
+                user_models.UserSettingsModel.get_all(
+                    include_deleted=False))
+            | 'Extract id and email' >> beam.Map(
+                    lambda user_setting: (
+                        user_setting.id, user_setting.email))
+        )
+
+        group_by_user_id = (
+            (collection_pair, user_pair)
+            | 'Group by user_id' >> beam.CoGroupByKey()
+            | 'Drop user id' >> beam.Values()  # pylint: disable=no-value-for-parameter
+            | 'Filter out results without any collection' >> beam.Filter(
+                lambda collection_email: len(collection_email[0]) > 0
+            )
+        )
+
+        return (
+            group_by_user_id
+            | 'Get final result' >> beam.MapTuple(
+                lambda collection, email: job_run_result.JobRunResult.as_stdout(
+                    'collection_ids: %s, email: %s' % (collection, email)
+                ))
+        )
+
+
+class FlattenIDs(beam.DoFn):  # type: ignore[misc]
+    """DoFn to extract user id and collection id."""
+
+    def process(
+        self, collection_rights_model: datastore_services.Model
+    ) -> Iterable[Tuple[str, str]]:
+        for user_id in collection_rights_model.owner_ids:
+            yield (user_id, collection_rights_model.id)
```

Comment:
This can be simplified into a function in `GetCollectionOwnersEmailsJob`

Reply:
Done